### PR TITLE
Remove direct use of Sirius solver (ANT-1152)

### PR DESCRIPTION
**Pull request by guilpier-code**
This PR intends to be a first work in order to remove the direct call of **Sirius** as a solver, **regarding the optimization weekly problem only**.
We want to keep the use of **Sirius**, but only called through **or-tools** interface.
The associated **gopro** ticket is [this one](https://gopro-tickets.rte-france.com/browse/ANT-1152).
Here we intend to : 
- remove the use of command line option **--use-ortools**
- rename current command line option **--ortools-solver** into **--solver**
- handle direct consequences of these changes.
- adapt legacy GUI (especially remove **use ortools** check box)

So this PR does not intend to be thorough : these first changes leave remaining dead code to be removed 
and renaming to be done. This further changes will be done in a coming PR.

**Remaining to be done** : 
- or-tools (even if we use **Sirius** behind it) does not print correctly MPS for named problems : optimization variables names are composed with time steps, set once for all (0, ..., 167) before running any year, and not updated at each week (we could expect step numbers to be inside [0, ..., 8759] depending on the week). 
  So we left the code that print these correctly. We'll have to find a solution for this.
  We could leave the responsibility of printing these named MPS to Antares code (as it did when Sirius was run alone) for a moment
- Banish all direct calls to **Sirius** from **Antares Simulator** (calls to **Sirius** are done in **hydro heuristics**, **adequacy patch CSR** and annual **quadratic** post-treatments of flows) does not seem to be suitable and urgent for now.
- What else ?

### CODE REVIEW

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp [hunk 1]
```diff
@@ -107,17 +105,13 @@ static SimplexResult OPT_TryToCallSimplex(const OptimizationOptions& options,
     {
         if (problemeHebdo->ReinitOptimisation)
         {
-            if (options.ortoolsUsed && solver)
+            if (solver != nullptr)
```
Comment (a-zakir) [inline src/solver/optimisation/opt_appel_solveur_lineaire.cpp]:
```suggestion
            if (solver)
```

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp [hunk 2]
```diff
@@ -201,40 +184,27 @@ static SimplexResult OPT_TryToCallSimplex(const OptimizationOptions& options,
...
 
     auto mps_writer = mps_writer_factory.create();
     mps_writer->runIfNeeded(writer, filename);
 
     TimeMeasurement measure;
-    if (options.ortoolsUsed)
+
+    const bool keepBasis = (optimizationNumber == PREMIERE_OPTIMISATION);
+    solver = ORTOOLS_Simplexe(&Probleme, solver, keepBasis, options);
+    if (solver != nullptr)
```
Comment (a-zakir) [inline src/solver/optimisation/opt_appel_solveur_lineaire.cpp]:
```suggestion
    if (solver)
```

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp [hunk 3]
```diff
@@ -244,14 +214,10 @@ static SimplexResult OPT_TryToCallSimplex(const OptimizationOptions& options,
     {
         if (ProblemeAResoudre->ExistenceDUneSolution != SPX_ERREUR_INTERNE)
         {
-            if (options.ortoolsUsed && solver)
+            if (solver != nullptr)
```
Comment (a-zakir) [inline src/solver/optimisation/opt_appel_solveur_lineaire.cpp]:
```suggestion
            if (solver)
```

File: src/solver/optimisation/opt_liberation_problemes_simplexe.cpp [hunk 4]
```diff
@@ -52,19 +52,13 @@ void OPT_LiberationProblemesSimplexe(const OptimizationOptions& options,
     {
         for (int numIntervalle = 0; numIntervalle < nbIntervalles; numIntervalle++)
         {
-            auto ProbSpx = (PROBLEME_SPX*)(ProblemeAResoudre->ProblemesSpx[numIntervalle]);
             auto solver = (MPSolver*)(ProblemeAResoudre->ProblemesSpx[numIntervalle]);
 
-            if (options.ortoolsUsed && solver)
+            if (solver != NULL)
```
Comment (a-zakir) [inline src/solver/optimisation/opt_liberation_problemes_simplexe.cpp]:
```suggestion
            if (solver)
```

File: src/tests/run-study-tests/actions_on_study/study_run.py [hunk 5]
```diff
@@ -23,9 +22,7 @@ def run(self):
         solver_full_path = str(Path(self.solver_path).resolve())
 
         command = [solver_full_path, "-i", str(self.study_path)]
```
Comment (a-zakir) [inline src/tests/run-study-tests/actions_on_study/study_run.py]: rename solver_full_path, i think it refers to Antares Simulator

File: src/tests/run-study-tests/fixtures.py [hunk 6]
```diff
@@ -40,8 +40,8 @@ def resultsRemover(study_path):
     return results_remover(study_path)
 
 @pytest.fixture
-def simulation(study_path, solver_path, use_ortools, ortools_solver, named_mps_problems, parallel):
-    return study_run(study_path, solver_path, use_ortools, ortools_solver, named_mps_problems, parallel)
+def simulation(study_path, solver_path, solver_name, named_mps_problems, parallel):
+    return study_run(study_path, solver_path, solver_name, named_mps_problems, parallel)
```
Comment (a-zakir) [inline src/tests/run-study-tests/fixtures.py]: renzme solver_path

File: src/ui/simulator/application/study.h [hunk 7]
```diff
@@ -114,7 +114,7 @@ bool CheckIfInsideAStudyFolder(const AnyString& path, bool quiet = false);
 ** \param simuComments Comments for the simulation
 ** \param ignoreWarnings True if warnings can be silently ignored
 ** \param useOrtools True if ortools must be used by antares-solver
```
Comment (a-zakir) [inline src/ui/simulator/application/study.h]:
```suggestion
```

File: src/tests/cucumber/features/steps/common_steps/simulator_utils.py [hunk 8]
```diff
@@ -28,13 +28,11 @@ def init_simu(context):
...
-        command.append('--use-ortools')
-        if "ortools-solver" in context.config.userdata:
-            solver = context.config.userdata["ortools-solver"]
-        else:
-            solver = "sirius"
-        command.append('--ortools-solver=' + solver)
+    solver = "sirius"
+    if "ortools-solver" in context.config.userdata:
+        solver = context.config.userdata["ortools-solver"]
+    command.append('--ortools-solver=' + solver)
```
Comment (pet-mit) [inline src/tests/cucumber/features/steps/common_steps/simulator_utils.py]:
```suggestion
    if "solver" in context.config.userdata:
        solver = context.config.userdata["solver"]
    command.append('--solver=' + solver)
```

**Review by pet-mit — CHANGES_REQUESTED** [review]

File: src/tests/cucumber/features/steps/common_steps/simulator_utils.py (hunk 8 above)
Comment (guilpier-code) [reply src/tests/cucumber/features/steps/common_steps/simulator_utils.py]: Done

File: src/tests/run-study-tests/actions_on_study/study_run.py (hunk 5 above)
Comment (guilpier-code) [reply src/tests/run-study-tests/actions_on_study/study_run.py]: You're totally right.
Even in the cmake files, antares simulator target is named **antares-solver**, which is misleading in a way

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp (hunk 1 above)
Comment (guilpier-code) [reply src/solver/optimisation/opt_appel_solveur_lineaire.cpp]: Done

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp (hunk 2 above)
Comment (guilpier-code) [reply src/solver/optimisation/opt_appel_solveur_lineaire.cpp]: Done

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp (hunk 3 above)
Comment (guilpier-code) [reply src/solver/optimisation/opt_appel_solveur_lineaire.cpp]: Done

File: src/solver/optimisation/opt_liberation_problemes_simplexe.cpp (hunk 4 above)
Comment (guilpier-code) [reply src/solver/optimisation/opt_liberation_problemes_simplexe.cpp]: Done

File: src/tests/run-study-tests/actions_on_study/study_run.py (hunk 5 above)
Comment (guilpier-code) [reply src/tests/run-study-tests/actions_on_study/study_run.py]: Done

File: src/tests/run-study-tests/fixtures.py (hunk 6 above)
Comment (guilpier-code) [reply src/tests/run-study-tests/fixtures.py]: Done

File: src/ui/simulator/application/study.cpp [hunk 9]
```diff
@@ -1207,14 +1206,8 @@ void RunSimulationOnTheStudy(Data::Study::Ptr study,
...
-            // Use ortools ?
-            if (useOrtools)
-            {
-                cmd << " --use-ortools";
-
-                // add solver name for ortools
-                cmd << " --ortools-solver=" << ortoolsSolver;
-            }
+            // add solver name for ortools
+            cmd << " --ortools-solver=" << solverName;
```
Comment (pet-mit) [inline src/ui/simulator/application/study.cpp]:
```suggestion
            cmd << " --solver=" << solverName;
```

**Review by pet-mit — CHANGES_REQUESTED** [review]

File: src/ui/simulator/application/study.h (hunk 7 above)
Comment (guilpier-code) [reply src/ui/simulator/application/study.h]: Done

File: src/ui/simulator/application/study.cpp (hunk 9 above)
Comment (guilpier-code) [reply src/ui/simulator/application/study.cpp]: yes ok

File: src/tests/end-to-end/test_simple-test.py [hunk 10]
```diff
@@ -1,95 +0,0 @@
-import os
```
Comment (guilpier-code) [inline src/tests/end-to-end/test_simple-test.py:1]: No longer used. Was removed.

**Review by a-zakir — APPROVED** [review]

**Review by pet-mit — APPROVED** [review]

**Review by flomnes — CHANGES_REQUESTED** [review]
- Update command-line documentation
- Revert .yml workflow files

**Review by flomnes — APPROVED** [review]

**Pull request merged**